### DISCUSSION TITLE
TypeScript error overlay

### DISCUSSION THREAD

**Discussion by schoening** [remix-run/remix#2719] [Proposals]
When working with Webpack, Parcel, Vite or react-cosmos I use TypeScript plugins to display an error overlay. These can run in parallel so they do not slow anything down (esbuild and swc being much faster than tsc)
So after some miliseconds I get an overlay on my app if the TypeScript compiler found errors:

_(image is the first thing from google, there are many different cool looking error overlays)_
![image](https://user-images.githubusercontent.com/1976060/162633556-d661af61-d7a2-40df-8caf-20e43daff9db.png)

It would be awesome to get this feature in Remix. If TypeScript fails I do **NOT** want to be able to proceed working. If I disagree with the error I can tweak the tsconfig.json

**Comment by TheRealFlyingCoder** [comment]
This is a bigger discussion that's been had a couple times.

The current stance from the team has been "Remix doesn't ever check types, you have to manually do it yourself" due to the slow nature of `tsc`.

I personally think there should be some surfacing of type errors regardless of speed OOTB, because that's why we choose typescript, and it's a risk for unknowing users to work in a typescript project and never be prompted if they've made a mistake.

A solution that runs in parallel, either to output an overlay or feed into the same console would be ideal though... I don't mind if the TS errors come a couple seconds after the build completes.

**Comment by await-ovo** [comment]
Also want to display any compile failure errors  on error overlay, currently it only log errors on terminal:

![image](https://user-images.githubusercontent.com/41503212/201259007-ebaf30e6-98dc-4a73-846b-e75b63cd07c7.png)

and after the page is reloaded,  no error messages can be seen

**Comment by abecks** [comment]
Just started using Remix and ran into this pretty quickly. I guess we're a bit spoiled these days and come to expect something like this. Definitely improves the DX. Saves having to leave a console open and watch for errors. I understand deferring TypeScript errors. Build failure stack traces would be nice though.
